libraries/AP_HAL/examples/TetrisGame/TetrisGame.cpp

```diff
@@ -0,0 +1,382 @@
+//
+// Tetris Game APM v0.0.1
+//
+// Tetris console game example for APM
+// Author: Hiroshi Takey, November 2017
+//
+
+/**
+*   Description.- Tetris game example displaying ASCII render
+*   to console, it read input control from console and
+*   from RC Channel for move, block change and block speed.
+*
+*   Keyboard: 'a' = move left, 'd' = move right, ' ' = block change,
+*   's' = block speed
+*
+*   Joy RC: RC Roll = move left and right, RC Pitch UP = block change,
+*   RC Pitch Down = block speed
+*/
+
+#include "TetrisGame.h"
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+
+GameTetris gametetris;
+
+Tetris::tetris_block blocks[] = {
+    {{"##",    "##"},       2, 2},
+    {{" X ",   "XXX"},      3, 2},
+    {{"@@@@"},              4, 1},
+    {{"OO",    "O ", "O "}, 2, 3},
+    {{"&&",    " &", " &"}, 2, 3},
+    {{"ZZ ",   " ZZ"},      3, 2}
+};
+
+#define TETRIS_PIECES (sizeof(blocks)/sizeof(tetris_block))
+
+#define MAX_CHANNELS    4
+// Move block left right
+#define JOY_LEFTRIGHT   0
+// Change block and speed
+#define JOY_CHGDOWN     2
+
+static uint8_t max_channels = 0;
+static uint16_t last_value[MAX_CHANNELS];
+
+#define SCHED_TASK(func, _interval_ticks, _max_time_micros) SCHED_TASK_CLASS(GameTetris, &gametetris, func, _interval_ticks, _max_time_micros)
+
+/*
+  scheduler table - all regular tasks are listed here, along with how
+  often they should be called (in 20ms units) and the maximum time
+  they are expected to take (in microseconds)
+ */
+const AP_Scheduler::Task GameTetris::scheduler_tasks[] = {
+    SCHED_TASK(ins_update,              50,   1000),
```

Comment:
why do you need an IMU ? just for clock ?

Reply:
this is the standard use for an APM App vehicle, we can make without it but the purpose about this example is to help like an APM app template. And it's good to teach and learn the multiple use about the Scheduler for syncronized time with external imu.

Personally i like very much the APM scheduler use.
And maybe it can do more fun things with imu sensors on the next versions :)